### PR TITLE
B1.B Backend DELETE operations on Course Controller for Courses, Course Staff

### LINKED ISSUE #38: Issue B1.B Backend DELETE operations, Course Controller for Courses, Course Staff
- [ ] There is a DELETE endpoint /api/course/delete?id=123 that deletes a course but only if you are either (a) an admin, or (b) an instructor AND your user id appears in the courseStaff table for the course.

### PR DESCRIPTION
**Pull request by dayvidwang**
This PR adds a DELETE endpoint for courses along with associated tests. This endpoint is restricted to users only if they have the ADMIN role or if they have the INSTRUCTOR role and they are listed in the course staff for the targeted course.

Closes #38

### CODE REVIEW

**Review by darylou — APPROVED** [review]
LGTM!

**Review by shuyiwan — APPROVED** [review]
LGTM!

File: src/main/java/edu/ucsb/cs156/organic/controllers/CoursesController.java [hunk 1]
```diff
@@ -173,14 +173,45 @@ public Course updateCourse(
...
+            if (staff.getGithubId().equals(u.getGithubId())) {
+                isStaffOrAdmin = true;
+                break;
+            }
+        }
+
+        if (!isStaffOrAdmin) {
+            throw new AccessDeniedException(
+                    "User is not a staff member for this course");
+        }
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/organic/controllers/CoursesController.java]: Please see: https://ucsb-cs156.github.io/topics/sql/sql_queries.html for a better way to do this

**Review by pconrad — CHANGES_REQUESTED** [review]
Same issue as the other PR: 

https://ucsb-cs156.github.io/topics/sql/sql_queries.html

**Review by darylou — APPROVED** [review]
LGTM!

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**